api/task_check_run.go

```diff
@@ -77,6 +77,8 @@ const (
 	TaskCheckDatabaseConnect TaskCheckType = "bb.task-check.database.connect"
 	// TaskCheckInstanceMigrationSchema is the task check type for migrating schemas.
 	TaskCheckInstanceMigrationSchema TaskCheckType = "bb.task-check.instance.migration-schema"
+	// TaskCheckTimingEarliestAllowedTime is the task check type for earliest allowed time.
+	TaskCheckTimingEarliestAllowedTime TaskCheckType = "bb.task-check.timing.earliest-allowed-time"
```

Comment:
I would call it bb.task-check.task.earliest-allowed-time or bb.task-check.general.earliest-allowed-time since it's a task general check